test/common/router/route_fuzz_test.cc

```diff
@@ -48,22 +48,29 @@ cleanRouteConfig(envoy::config::route::v3::RouteConfiguration route_config) {
 bool validateMatcherConfig(const xds::type::matcher::v3::Matcher& matcher);
 
 bool validateOnMatchConfig(const xds::type::matcher::v3::Matcher::OnMatch& on_match) {
-  if (on_match.has_matcher() && !validateMatcherConfig(on_match.matcher())) {
-    return false;
+  switch (on_match.on_match_case()) {
+  case xds::type::matcher::v3::Matcher_OnMatch::kMatcher: {
+    return validateMatcherConfig(on_match.matcher());
   }
-  // Only envoy...Route is allowed as typed_config here.
-  if (on_match.action().typed_config().type_url().find("envoy.config.route.v3.Route") ==
-      std::string::npos) {
-    return false;
+  case xds::type::matcher::v3::Matcher_OnMatch::kAction: {
+    // Only envoy...Route is allowed as typed_config here.
+    if (on_match.action().typed_config().type_url().find("envoy.config.route.v3.Route") ==
+        std::string::npos) {
+      return false;
+    }
+    envoy::config::route::v3::Route on_match_route_action_config;
+    MessageUtil::unpackTo(on_match.action().typed_config(), on_match_route_action_config);
+    ENVOY_LOG_MISC(trace, "typed_config of on_match.action is: {}",
+                   on_match_route_action_config.DebugString());
+    return !isUnsupportedRouteConfig(on_match_route_action_config);
+  }
+  case xds::type::matcher::v3::Matcher_OnMatch::ON_MATCH_NOT_SET: {
+    // This alternative shall never occur. It is here to case on all values of Matcher_OnMatch and
+    // this way ensures that the compiler will report new entries to that enum as warning, which
+    // will finally result in an compiler error, when error on all warnings is set.
+    ASSERT(false);
```

Comment:
I am not sure if here will work as you explain.  Let's say have new enum `kNew`,  this `xds::type::matcher::v3::Matcher_OnMatch::ON_MATCH_NOT_SET` will not catch it , right?

I think you will need the assert at the end out side of all three cases? something like below : 
switch (on_match.on_match_case()) {
   case kMatcher: {
   .......
   } 
   case kAction: {
   .......
   }
  case ON_MATCH_NOT_SET: {
  .....
  }
  }
  PANIC_DUE_TO_CORRUPT_ENUM;

Reply:
That is only when warnings in clang is enabled, right?
Even though envoy enable most warnings, I think it is better to program in a generic way-- i.e., compiler agnostic or compiler setting agnostic way

Reply:
changed